scale-tests/streaming_test.py

```diff
@@ -0,0 +1,101 @@
+import json
+import sys
+
+import sdk_cmd
+import shakedown
+import spark_utils as utils
+from tests import test_kafka
+
+KAFKA_PACKAGE_NAME = "kafka"
+KAFKA_SERVICE_NAME = "kafka"
+PRODUCER_SERVICE_NAME = "Spark->Kafka Producer"
+
+
+def run_pipeline(kerberos_flag, num_consumers, stop_count, jar_uri, spark_app_name):
+    stop_count = str(stop_count)
+    broker_dns = _kafka_broker_dns()
+    topic = "top1"
+
+    big_file, big_file_url = "file:///mnt/mesos/sandbox/big.txt", "http://norvig.com/big.txt"
+
+    # arguments to the application
+    producer_args = " ".join([broker_dns, big_file, topic, kerberos_flag])
+
+    uris = "spark.mesos.uris=http://norvig.com/big.txt"
+
+    common_args = [
+        "--conf", "spark.mesos.containerizer=mesos",
+        "--conf", "spark.scheduler.maxRegisteredResourcesWaitingTime=2400s",
+        "--conf", "spark.scheduler.minRegisteredResourcesRatio=1.0",
+        "--conf", uris
+    ]
+
+    producer_config = ["--conf", "spark.cores.max=2", "--conf", "spark.executor.cores=2",
+                       "--class", "KafkaFeeder"] + common_args
+
+    producer_id = utils.submit_job(app_url=jar_uri,
+                                   app_args=producer_args,
+                                   app_name=spark_app_name,
+                                   args=producer_config,
+                                   verbose=False)
+
+    shakedown.wait_for(lambda: test_kafka._producer_launched(), ignore_exceptions=False, timeout_seconds=600)
+    # shakedown.wait_for(lambda: utils.is_service_ready(KAFKA_SERVICE_NAME, 1),
+    #                   ignore_exceptions=False, timeout_seconds=600)
+
+    consumer_config = ["--conf", "spark.cores.max=4", "--class", "KafkaConsumer"] + common_args
+
+    consumer_args = " ".join([broker_dns, topic, stop_count, kerberos_flag])
+
+    for i in range(0, num_consumers):
+        utils.run_tests(app_url=jar_uri,
+                        app_args=consumer_args,
+                        expected_output="Read {} words".format(stop_count),
+                        app_name=spark_app_name,
+                        args=consumer_config)
+
+    utils.kill_driver(producer_id, spark_app_name)
```

Comment:
I've noticed this line is a little flaky. Sometimes I see the producer doesn't get killed in the Soak cluster, it just runs forever. I don't know of a quick fix, except to modify the Producer app and add a "stop_count" parameter like in the Consumer. For now, I guess it won't be an issue because the Producer would get killed anyway when the dispatchers are uninstalled.

Reply:
Decided to leave it out bc like you said, uninstalling dispatchers shuts down producers. 2 for 1 deal :)